lib/mirage.ml

```diff
@@ -190,15 +190,15 @@ let stdlib_random = impl stdlib_random_conf
 
 let query_ocamlfind ?(recursive = false) ?(format="%p") ?predicates libs =
   let open Bos in
-  let flag = if recursive then "-recursive" else ""
+  let flag = if recursive then [ "-recursive" ] else []
   and format = Cmd.of_list [ "-format" ; format ]
   and predicate = match predicates with
     | None -> []
     | Some x -> [ "-predicates" ; x ]
   and q = "query"
   in
   let cmd =
-    Cmd.(v "ocamlfind" % q % flag %% format %% of_list predicate %% of_list libs)
+    Cmd.(v "ocamlfind" % q %% of_list flag %% format %% of_list predicate %% of_list libs)
```

Comment:
IIRC I needed this because with `recursive=false` it was passing an empty string argument to `ocamlfind` instead of omitting it, but I'm not very confident about all the operators here. Let me know if it looks suss...